src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt

```diff
@@ -203,72 +203,79 @@ class OrganizationMetadataEntitySetsService(private val edmService: EdmManager)
         )
     }
 
-    fun addDatasetColumn(entitySet: EntitySet, propertyType: PropertyType) {
+    fun addDatasetColumn(entitySet: EntitySet, propertyTypesInEntitySet: Collection<PropertyType>) {
         initializeFields()
         if (!isFullyInitialized()) {
             return
         }
         val organizationId = entitySet.organizationId
         val organizationMetadataEntitySetIds = organizationService.getOrganizationMetadataEntitySetIds(organizationId)
-        val columnEntity = mutableMapOf<UUID, Set<Any>>(
-                propertyTypes.getValue(ID).id to setOf(propertyType.id.toString()),
-                propertyTypes.getValue(DATASET_NAME).id to setOf(entitySet.name),
-                propertyTypes.getValue(COL_NAME).id to setOf(propertyType.type.fullQualifiedNameAsString),
-                propertyTypes.getValue(ORG_ID).id to setOf(organizationId.toString()),
-                propertyTypes.getValue(TYPE).id to setOf(propertyType.datatype.toString()),
-                propertyTypes.getValue(DESCRIPTION).id to setOf(propertyType.description)
-        )
-        val datasetColumnEntity = mutableMapOf<UUID, Set<Any>>(
-                propertyTypes.getValue(COL_INFO).id to setOf(mapper.writeValueAsString(propertyType))
-        )
 
-        val columnEntityKeyId = getColumnEntityKeyId(organizationMetadataEntitySetIds, entitySet.id, propertyType)
-        val datasetEntityKeyId = getDatasetEntityKeyId(organizationMetadataEntitySetIds, entitySet.id)
+        val propertyTypeEntities = propertyTypesInEntitySet.map {propertyType ->
+            val columnEntity = mutableMapOf<UUID, Set<Any>>(
+                    propertyTypes.getValue(ID).id to setOf(propertyType.id.toString()),
+                    propertyTypes.getValue(DATASET_NAME).id to setOf(entitySet.name),
+                    propertyTypes.getValue(COL_NAME).id to setOf(propertyType.type.fullQualifiedNameAsString),
+                    propertyTypes.getValue(ORG_ID).id to setOf(organizationId.toString()),
+                    propertyTypes.getValue(TYPE).id to setOf(propertyType.datatype.toString()),
+                    propertyTypes.getValue(DESCRIPTION).id to setOf(propertyType.description)
+            )
 
+            val columnEntityKeyId = getColumnEntityKeyId(organizationMetadataEntitySetIds, entitySet.id, propertyType)
+            dataGraphManager.partialReplaceEntities(
```

Comment:
I think we should probably pull this out of the forEach, so like have the forEach construct the map of ekids to entities, and then do one batch upsert using `partialReplaceEntities`

Reply:
good idea, much cleaner too.